exo-go/src/app_runner/index.go

```
@@ -0,0 +1,145 @@
+package appRunner
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"sync"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/Originate/exosphere/exo-go/src/docker_compose"
+	"github.com/Originate/exosphere/exo-go/src/docker_helpers"
+	"github.com/Originate/exosphere/exo-go/src/logger"
+	"github.com/Originate/exosphere/exo-go/src/process_helpers"
+	"github.com/Originate/exosphere/exo-go/src/service_config_helpers"
+	"github.com/Originate/exosphere/exo-go/src/service_helpers"
+	"github.com/Originate/exosphere/exo-go/src/types"
+	"github.com/docker/docker/client"
+	"github.com/fatih/color"
+	"github.com/pkg/errors"
+)
+
+// AppRunner runs the overall application
+type AppRunner struct {
+	AppConfig            types.AppConfig
+	Logger               *logger.Logger
+	Env                  []string
+	DockerConfigLocation string
+	Cwd                  string
+	OnlineTexts          map[string]string
+}
+
+// NewAppRunner is AppRunner's constructor
+func NewAppRunner(appConfig types.AppConfig, logger *logger.Logger) *AppRunner {
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("Failed to get the current path: %s", err)
+	}
+	appRunner := &AppRunner{AppConfig: appConfig, Logger: logger, DockerConfigLocation: path.Join(cwd, "tmp"), Cwd: cwd}
+	for _, dependency := range appConfig.Dependencies {
+		for variable, value := range dependency.GetEnvVariables() {
+			appRunner.Env = append(appRunner.Env, fmt.Sprintf("%s=%s", variable, value))
+		}
+	}
+	return appRunner
+}
+
+// Start runs the application
+func (appRunner *AppRunner) Start() {
+	_, stdoutBuffer, err := dockerCompose.RunAllImages(appRunner.Env, appRunner.DockerConfigLocation, appRunner.Write)
+	if err != nil {
+		appRunner.Shutdown("", "Failed to run images")
+	} else {
+		appRunner.compileOnlineText(func(err error) {
+			if err != nil {
+				log.Fatal(err)
```

Since these are helper functions lets return errors instead of using log.Fatal